src/main/scala/cromwell/engine/SymbolStore.scala

```diff
@@ -26,9 +26,46 @@ class SymbolStore(binding: WdlBinding, inputs: Map[FullyQualifiedName, WdlValue]
     assignSymbolStoreEntry(fullyQualifiedName, value, input = true)
   }
 
+  binding.workflows foreach {workflow =>
+    workflow.calls foreach {call =>
+      call.inputMappings foreach {
+        case (k, v) => assignSymbolStoreEntry(s"${call.fullyQualifiedName}.$k", v, input = true)
+      }
+    }
+  }
+
   def locallyQualifiedInputs(call: Call): Map[String, WdlValue] = {
+
+    def lookup(s: String): WdlValue = {
+      var workflow = call.parent.getOrElse {
+        throw new WdlExpressionException("Expecting 'call' to have a 'workflow' parent")
+      }.asInstanceOf[Workflow]
+      val matchedCall = workflow.calls.find{c => c.name == s}.getOrElse {
+        throw new WdlExpressionException(s"Expecting to find a call with name '$s'")
+      }
+      val callOutputs = store.collect{ case entry if entry.scope == matchedCall.fullyQualifiedName && entry.isOutput =>
+          val value = entry.wdlValue match {
+            case Some(v) => v
+            case _ => throw new WdlExpressionException(s"Could not evaluate call '${matchedCall.name}', because '${entry.key.name}' is undefined")
+          }
+          entry.key.name -> value
+      }.toMap[String, WdlValue]
+      WdlObject(callOutputs)
+    }
+
+    object CallInputWdlFunctions extends WdlFunctions {
+      def getFunction(name: String): WdlFunction = {
+        throw new WdlExpressionException("TODO: Some functions may be allowed in this context")
+      }
+    }
+
     store.collect { case entry if entry.scope == call.fullyQualifiedName && entry.isInput =>
-      entry.key.name -> entry.wdlValue.get
+      val value = entry.wdlValue match {
```

Comment:
similar comment as the collect above. I think these constructs should read as close to an english sentence as possible on a single line, and anything which prevents that should be abstracted out to a func if possible